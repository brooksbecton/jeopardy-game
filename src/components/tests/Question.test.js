import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";

import Question from "./../Question";

const mockProps = { question: { question: "Is this the real life?" } };

beforeAll(() => {
  configure({ adapter: new Adapter() });
});

describe("Question Component", () => {
  it("renders correctly", () => {
    const component = shallow(<Question {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
  it("has class 'question'", () => {
    const component = shallow(<Question {...mockProps} />);
    expect(component.first("div").hasClass("question")).toBeTruthy();
  });
  it("displays question prop", () => {
    const component = shallow(<Question {...mockProps} />);
    expect(component.text()).toBe("Is this the real life?");
  });
});
