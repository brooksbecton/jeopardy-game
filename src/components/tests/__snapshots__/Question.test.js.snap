// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    question={
      Object {
        "question": "Is this the real life?",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "Is this the real life?",
      "className": "question",
    },
    "ref": null,
    "rendered": "Is this the real life?",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Is this the real life?",
        "className": "question",
      },
      "ref": null,
      "rendered": "Is this the real life?",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
